tests/BuildScriptGenerator.Tests/Node/NodePlatformTest.cs

```diff
@@ -160,6 +161,122 @@ public void BuildScript_DoesNotHaveSdkInstallScript_IfDynamicInstallNotEnabled_A
             Assert.Null(buildScriptSnippet.PlatformInstallationScriptSnippet);
         }
 
+        [Fact]
+        public void GeneratedBuildSnippet_DoesNotThrowException_IfPackageJsonHasBuildNode_AndRequireBuildPropertyIsSet()
+        {
+            // Arrange
+            const string packageJson = @"{
+              ""main"": ""server.js"",
+              ""scripts"": {
+                ""build"": ""build-node"",
```

Comment:
nit: it may sound crazy, but your package.json is simple string. can we send build and build:azure as parameter to this test and make it [theory] instead of two different [fact] uts

Reply:
Yeah I had considered it initially but I created separate cases because they are 2 different pivots and also since these are unit tests, we also need not optimize for performance over pivot separation.